x-pack/filebeat/input/entityanalytics/provider/azuread/azure.go

```diff
@@ -321,7 +370,25 @@ func (p *azure) doFetch(ctx context.Context, state *stateStore, fullSync bool) (
 		})
 	})
 
-	return updatedUsers, nil
+	// Expand device group memberships.
+	updatedDevices.ForEach(func(devID uuid.UUID) {
+		d, ok := state.devices[devID]
+		if !ok {
+			p.logger.Errorf("Unable to find device %q in state", devID)
+			return
+		}
+		d.Modified = true
+		if d.Deleted {
+			return
```

Comment:
A debug log to say control exited from `d.Deleted` block ?

Reply:
@taylor-swanson?